src/accommodation/accommodation.service.ts

```diff
@@ -346,23 +346,41 @@ export class AccommodationService {
     const result =
       (!checkIn && checkOut) ||
       (checkIn && !checkOut) ||
-      (checkIn && checkOut && dayjs(checkOut).isSameOrBefore(dayjs(checkIn)));
+      (checkIn && checkOut && dayjs(checkOut).isSameOrBefore(dayjs(checkIn), 'day')) ||
```

Comment:
if you have such new condition than you can return falsy result erlier
if (!checkIn || !checkOut) return false
return dayjs(checkOut).isSameOrBefore(dayjs(checkIn), 'day')) || dayjs(checkIn).isBefore(dayjs(), 'day')) || dayjs(checkOut).isSameOrBefore(dayjs(), 'day'));